projects/kit/components/input-number/input-number.component.ts

```diff
@@ -318,6 +313,12 @@ export class TuiInputNumberComponent
         return maskitoParseNumber(this.nativeValue, this.numberFormat.decimalSeparator);
     }
 
+    private get precision(): number {
+        return Number.isNaN(this.numberFormat.decimalLimit)
```

Comment:
When `decimalLimit` could be equal to `NaN` ?
Could we use `Infinity` instead `NaN`

Reply:
NaN is the default value. We use it to set different default values in different components